### DISCUSSION TITLE
Create a listings site based on the content of a folder with generated revealjs slides ( or their pdfs)

### DISCUSSION THREAD

**Discussion by tomkom** [quarto-dev/quarto-cli#6236] [Q&A]
### Description

I am using the approach from #1433 to generate slide content ( revealjs) based on a subset of the content of my book. I would like to generate a listings page from this content. I have a folder _slides/ where all the revealjs generated content lives (hence, these are html sites and associated resources). I also have a folder where pdfs exist. But there is no folder with qmd sources, as these are subsets ( based on ':::{.content-visible when-format="revealjs"}' and ':::{.content-visible unless-format="revealjs"}'. I am wondering how one could generate the listings... I read #5849 but I am no wiser.

Thanks!

**Comment by mcanouil** [comment]
If you generate the contents, you can generate the listing based on a yaml file, see <https://quarto.org/docs/websites/website-listings.html> for various type of contents (including yaml, globbing, hardcoded files, etc.)

```qmd
---
listing:
  contents:
    - my-listing-content.yml
```

**Comment by tomkom** [comment]
This would require the scripted generation of the yml, right? Would ot work
for html contents (revealjs slides)?

**Reply by mcanouil** [reply]
Yes, it would, but you can achieve that using a pre-render script, see <https://quarto.org/docs/projects/scripts.html#pre-and-post-render>.

There are several examples of pre-render scripts in various GitHub Discussions here.
